mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -105,6 +110,12 @@ class RegressionMetrics @Since("1.2.0") (
    */
   @Since("1.2.0")
   def r2: Double = {
-    1 - SSerr / SStot
+    // In case of regression through the origin (biased case), the definition of R^2 is
+    // to be modified. See jira SPARK-12331
```

Comment:
Instead of referencing the JIRA in the comment, it is probably better to explicitly define how the value is computed in the scaladoc. A reference to an appropriate paper might be good as well.